ObjectiveGit/GTTreeEntry.m

```diff
@@ -39,6 +39,7 @@
 
 @interface GTTreeEntry ()
 @property (nonatomic, assign, readonly) const git_tree_entry *git_tree_entry;
+@property (nonatomic, assign, readonly) git_tree_entry* entry_to_free;
```

Comment:
I find this a little fishy, so I'll let the people in the know give their opinion ;-).

I *think* the `const` on the line above should be dropped so `GTTreeEntry` assumes ownership of the underlying `git_tree_entry`, then this duplication becomes unnecessary.